dex/farm-with-locked-rewards/src/lib.rs

```diff
@@ -163,7 +162,10 @@ pub trait Farm:
             "Exit amount is bigger than the payment amount"
         );
 
-        let boosted_rewards_full_position = self.claim_only_boosted_payment(&orig_caller, &payment);
+        let boosted_rewards = self.claim_only_boosted_payment(&orig_caller);
```

Comment:
Call migrate_old_farm_position here as well.

Reply:
claim rewards and make a new position for the whole payment.

Do the same on farm-staking.